specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/containerregistry_build.json

```diff
@@ -0,0 +1,2643 @@
+          "enum": [
+            "Default",
+            "Token"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "UpdateTriggerPayloadType",
+            "modelAsString": true
+          }
+        },
+        "status": {
+          "description": "The current status of trigger.",
+          "default": "Enabled",
+          "enum": [
+            "Disabled",
+            "Enabled"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "TriggerStatus",
+            "modelAsString": true
+          }
+        },
+        "name": {
+          "description": "The name of the trigger.",
+          "type": "string"
+        }
+      }
+    },
+    "SourceUpdateParameters": {
+      "description": "The properties for updating the source code repository.",
+      "type": "object",
+      "properties": {
+        "sourceControlType": {
+          "description": "The type of source control service.",
+          "enum": [
+            "Github",
+            "VisualStudioTeamService"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "SourceControlType",
+            "modelAsString": true
+          }
+        },
+        "repositoryUrl": {
+          "description": "The full URL to the source code repository",
+          "type": "string"
+        },
+        "branch": {
+          "description": "The branch name of the source code.",
+          "type": "string"
+        },
+        "sourceControlAuthProperties": {
+          "$ref": "#/definitions/AuthInfoUpdateParameters",
+          "description": "The authorization properties for accessing the source code repository and to set up\r\nwebhooks for notifications."
+        }
+      }
+    },
+    "AuthInfoUpdateParameters": {
+      "description": "The authorization properties for accessing the source code repository.",
+      "type": "object",
+      "properties": {
+        "tokenType": {
+          "description": "The type of Auth token.",
+          "enum": [
+            "PAT",
+            "OAuth"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "TokenType",
+            "modelAsString": true
+          }
+        },
+        "token": {
+          "description": "The access token used to access the source control provider.",
+          "type": "string"
+        },
+        "refreshToken": {
+          "description": "The refresh token used to refresh the access token.",
+          "type": "string"
+        },
+        "scope": {
+          "description": "The scope of the access token.",
+          "type": "string"
+        },
+        "expiresIn": {
+          "format": "int32",
+          "description": "Time in seconds that the token remains valid",
+          "type": "integer"
+        }
+      }
+    },
+    "Resource": {
+      "description": "An Azure resource
```

Comment:
This extension is a Mapper for mapping the value on the wire to the model definition defined in the specification. Some services have the value of the discriminator property as `"Microsoft.TaskRunRequest"` whereas the model definition name in the swagger spec is `"TaskRunRequest"`. 

In your case, the name of the model and the value provided in the `"x-ms-discriminator-value"` is the same. Thus please remove this extension from all the places in the spec where it's value and the name of the model are the same.